index.js

```diff
@@ -17,11 +18,24 @@ CoffeeScriptFilter.prototype.targetExtension = 'js'
 CoffeeScriptFilter.prototype.processString = function (string, srcFile) {
   var coffeeScriptOptions = {
     bare: this.bare,
-    literate: coffeeScript.helpers.isLiterate(srcFile)
+    literate: coffeeScript.helpers.isLiterate(srcFile),
+    sourceMap: !!this.sourceMap,
+
+    sourceFiles: [ srcFile ],
```

Comment:
Wouldn't `path.basename` just going to flatten all the files? If there are 2 files of the same name in different directories, they will be displayed as the same file.